src/core/memory.d

```diff
@@ -810,25 +810,31 @@ struct GC
     }
 }
 
-/// Pure variant of `malloc` that doesn't affect global variable `errno`.
+/**
+ * Pure variants of C's memory allocation functions. Purity is achieved via
+ * resetting the `errno` to it's value prior to being called, removing the
+ * function's global state mutation.
+ *
+ * See_Also:
+ *     $(LINK2 https://dlang.org/spec/function.html#pure-functions, D's rules for purity),
+ *     which allow for memory allocation under specific circumstances.
+ */
 void* pureMalloc(size_t size) @trusted pure
 {
     const errno = fakePureGetErrno();
     void* ret = fakePureMalloc(size);
     cast(void)fakePureSetErrno(errno);
```

Comment:
I'd say it's more efficient to guard the call like this:

```d
if (!ret || errno != ERR_OK) cast(void)fakePureSetErrno(errno);
```

The common case is malloc succeeds and the previous state was ERR_OK so no need to reset errno.